src/h5web/visualizations/heatmap/utils.ts

```diff
@@ -0,0 +1,28 @@
+import { rgb } from 'd3-color';
+import { scaleSequential } from 'd3-scale';
+import { interpolateMagma } from 'd3-scale-chromatic';
+
+export function computeTextureData(matrix: number[][]): Uint8Array {
+  // Flatten input matrix
+  const values = matrix.flat();
+
+  // Compute domain
+  const domain: [number, number] = values.reduce(
```

Comment:
There is no dedicated function to get the min/max of an array ?

Reply:
Nope, but perhaps in lodash, I'll have a look.

Reply:
There is indeed [_.min](https://lodash.com/docs/4.17.15#min) and [_.max](https://lodash.com/docs/4.17.15#max) but that means traversing the array twice to get both so I think your implementation is better.

Reply:
There's `min` and `max` in lodash, but not both together. We'd get O(2n) instead of O(n). Not awful, but significant nonetheless. I found two libraries that could work instead, notably https://github.com/dy/array-bounds, but neither of them provides type declarations. 😞 

I opened an issue on the lodash repository; I thought it was something worth suggesting: https://github.com/lodash/lodash/issues/4711